diff --git a/dist/tools/trade.js b/dist/tools/trade.js
index 280aa7564613238857477f791f7f7522a8faca83..47ad1ffee7884cb6a2e19e00d410fb2677ad8e2c 100644
--- a/dist/tools/trade.js
+++ b/dist/tools/trade.js
@@ -29,7 +29,7 @@ async function trade(agent, outputMint, inputAmount, inputMint = constants_1.TOK
             `&amount=${scaledAmount}` +
             `&slippageBps=${slippageBps}` +
             `&onlyDirectRoutes=true` +
-            `&maxAccounts=20` +
+            `&maxAccounts=64` +
             `${agent.config.JUPITER_FEE_BPS ? `&platformFeeBps=${agent.config.JUPITER_FEE_BPS}` : ""}`)).json();
         // Get serialized transaction
         let feeAccount;
